server/src/api/v1.py

```diff
@@ -489,6 +491,38 @@ def agents_post(agent_name, json_data):
     return "OK"
 
 
+@v1.post("/agents/status")
+@v1.input(schemas.StatusUpdate, location="json")
+def agents_status_post(json_data):
+    """Posts status updates from the agent to the server to be forwarded
+    to TestObserver
+
+    The json sent to this endpoint may contain data such as the following:
+    {
+        "agent_id": "<string>",
+        "job_queue": "<string>",
+        "job_status_webhook": "<URL as string>",
+        "events": [
+        {
+            "event_name": "<string enum of events>",
+            "timestamp": "<datetime>",
+            "detail_msg": "<string>"
+        },
+        ...
+        ]
+    }
+
+    """
+    request_json = json_data
+    webhook_url = request_json.pop("job_status_webhook")
+    try:
+        # Times out after 1 seconds
+        response = requests.post(webhook_url, json=request_json, timeout=1)
+        return response.text, response.status_code
```

Comment:
doesn't sound right to have a 1s timeout, I'd like to push again towards a queue

Reply:
Alternatively, if we can do something similar to [background tasks](https://fastapi.tiangolo.com/reference/background/) in fastapi

Reply:
I think for now we'll do a 3 second timeout with 3 retries. With regards to queues, there was some direction to use Temporal over Celery, but setting up that system might be better to do in a separate issue.

Reply:
sure, as long as we do go back to it. "for now" solutions tend to become "forever" :sweat_smile: